src/types/plugin-input.ts

```diff
@@ -24,10 +26,33 @@ export const startStopSchema = T.Object({
   ),
   miscellaneous: T.Object(
     {
-      maxConcurrentTasks: T.Number(),
       startRequiresWallet: T.Boolean(),
+      maxConcurrentTasks: T.Array(
+        T.Object({
+          role: userRoleSchema,
+          limit: T.Integer(),
+        })
+      ),
     },
-    { default: { maxConcurrentTasks: 3, startRequiresWallet: true } }
+    {
+      default: {
+        maxConcurrentTasks: [
```

Comment:
This default is not proper since `maxConcurrentTask` should have its own default, not inside the parent object.

Reply:
@jordan-ae this has not been taken care of. If the configuration looks like
```yml
        miscellaneous:
          # maxConcurrentTasks: []
          startRequiresWallet: true
```
It crashes with the message
```
        "Error: Unable to decode value as it does not match the expected schema"
```
because no default value is set. Also, it seems to crash with an empty array.